src/js/index.js

```diff
@@ -5,15 +5,29 @@ import '@/assets/scss/base.scss';
 import { getCharactersAll } from '@/js/services/characterService.js';
 import renderCharacters from './views/renderCharacter';
 import { getSelectedFilterRole, bindCharacterFilterRole } from './bindings/role';
+import { bindCharacterFilterPublisher, getSelectedFilterPublisher } from './bindings/publisher';
 
-const characterFilterOrSortingUpdated = (value, characters) => {
-  const filteredList = getSelectedFilterRole(value, characters);
-  document.getElementById('hero-list').innerHTML = '';
-  renderCharacters(filteredList);
-};
+// filter characters by publishers
+export function getFilteredCharactersByPublisher(characters, publishers) {
+  // eslint-disable-next-line max-len
+  // create a copy of characters array and filter by character then return a list of characters who have a publisher value.
+  // eslint-disable-next-line max-len
+  const filteredCharactersByPublisher = [...characters].filter((character) => publishers.includes(character.biography.publisher));
```

Comment:
Again lets get rid of these `eslint-disable-next-line max-len ` if you can. For line 15 what you can do to break if up into two line is move part of the function on to the next line like this.
```
  const filteredCharactersByPublisher = [...characters]
        .filter((character) => publishers.includes(character.biography.publisher));
```